source/source.go

```diff
@@ -0,0 +1,155 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-dropbox/config"
+	"github.com/conduitio-labs/conduit-connector-dropbox/pkg/dropbox"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+var (
+	ErrSourceClosed = fmt.Errorf("error source not opened for reading")
+	ErrReadingData  = fmt.Errorf("error reading data")
+)
+
+type Source struct {
+	sdk.UnimplementedSource
+
+	config    Config
+	position  *Position
+	client    dropbox.FoldersClient
+	ch        chan opencdc.Record
+	workersWg *sync.WaitGroup
+}
+
+type Config struct {
+	sdk.DefaultSourceMiddleware
+	config.Config
+
+	// Timeout for Dropbox longpolling requests.
+	LongpollTimeout time.Duration `json:"longpollTimeout" default:"30s"`
+	// Size of a file chunk in bytes to split large files, maximum is 4MB.
+	FileChunkSizeBytes uint64 `json:"fileChunkSizeBytes" default:"3145728"`
+	// Maximum number of retry attempts.
+	Retries int `json:"retries" default:"0"`
+	// Delay between retry attempts.
+	RetryDelay time.Duration `json:"retryDelay" default:"10s"`
+	// Maximum number of entries to fetch in a single list_folder request.
+	Limit int `json:"limit" default:"1000"`
+}
+
+func NewSource() sdk.Source {
+	return sdk.SourceWithMiddleware(&Source{})
```

Comment:
We should disable schema extraction, as we're working with raw data. I think the file connector is doing that too.